GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -364,12 +365,30 @@ protected virtual bool TryFetchReleases(out List<Release> releases, out string e
             return false;
         }
 
-        protected virtual void RunInstaller(string path, string args, out int exitCode, out string error)
+        protected virtual void RunInstaller(string path, string args, string certCN, out int exitCode, out string error)
         {
-            ProcessResult processResult = ProcessHelper.Run(path, args);
+            Stream stream = this.fileSystem.OpenFileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, false);
 
-            exitCode = processResult.ExitCode;
-            error = processResult.Errors;
+            if (stream == null)
+            {
+                exitCode = -1;
+                error = $"Could not launch installer {path}. Error reading file.";
+                return;
+            }
+
+            if (this.VerifyCodeSignature(path, certCN, out error))

```

Comment:
I generally prefer seeing `if (unusual situation) { return with error }` instead of `if (typical) { do stuff } else {return with error}`.